crates/goose/src/providers/anthropic.rs

```diff
@@ -0,0 +1,430 @@
+                    let name = block.get("name").and_then(|n| n.as_str())
+                        .ok_or_else(|| anyhow!("Missing tool_use name"))?;
+                    let input = block.get("input")
+                        .ok_or_else(|| anyhow!("Missing tool_use input"))?;
+                    
+                    let tool_call = ToolCall::new(name, input.clone());
+                    message = message.with_tool_request(id, Ok(tool_call));
+                }
+                _ => continue,
+            }
+        }
+
+        Ok(message)
+    }
+
+    async fn post(&self, payload: Value) -> Result<Value> {
+        let url = format!(
+            "{}/v1/messages",
+            self.config.host.trim_end_matches('/')
+        );
+
+        let response = self
+            .client
+            .post(&url)
+            .header("x-api-key", &self.config.api_key)
+            .header("anthropic-version", "2023-06-01")
+            .json(&payload)
+            .send()
+            .await?;
+
+        match response.status() {
+            StatusCode::OK => Ok(response.json().await?),
+            status if status == StatusCode::TOO_MANY_REQUESTS || status.as_u16() >= 500 => {
+                Err(anyhow!("Server error: {}", status))
+            }
+            status => {
+                let error_text = response.text().await?;
+                Err(anyhow!(
+                    "Request failed: {} - {}",
+                    status,
+                    error_text
+                ))
+            }
+        }
+    }
+}
+
+#[async_trait]
+impl Provider for AnthropicProvider {
+    async fn complete(
+        &self,
+        system: &str,
+        messages: &[Message],
+        tools: &[Tool],
+    ) -> Result<(Message, Usage)> {
+        let anthropic_messages = Self::messages_to_anthropic_spec(messages);
+        let tool_specs = Self::tools_to_anthropic_spec(tools);
+
+        // Check if we have any messages to send
+        if anthropic_messages.is_empty() {
+            return Err(anyhow!("No valid messages to send to Anthropic API"));
+        }
+
+        let mut payload = json!({
+            "model": self.config.model,
+            "messages": anthropic_messages,
+            "max_tokens": self.config.max_tokens.unwrap_or(4096)
+        });
+
+        // Add system message if present
+        if !system.is_empty() {
+            payload.as_object_mut().unwrap()
+                .insert("system".to_string(), json!(system));
+        }
+
+        // Add tools if present
+        if !tool_specs.is_empty() {
+            payload.as_object_mut().unwrap()
+                .insert("tools".to_string(), json!(tool_specs));
+        }
+
+        // Add temperature if specified
+        if let Some(temp) = self.config.temperature {
+            payload.as_object_mut().unwrap()
+                .insert("temperature".to_string(), json!(temp));
+        }
+
+        // Make request
+        let response = self.post(payload).await?;
```

Comment:
similar to openai provider, can we check for context limit exceeded here? i think we'll want to raise a specific error for this which then gets handled by the agent / shown to the user. 

https://github.com/block/goose/pull/401/files#diff-fed54b47734f06ba118bdf144c00ad3caeab4026f0d230997cb4a03c7d6be6e6